### PR TITLE
Fix stage reorder tests

### PR DESCRIPTION
**Pull request by eddie-c-davis**
## Technical Description

Refactors the stage reordering pass tests to only use IIR and not go through SIR.

### Resolves / Enhances

Work on #702

### CODE REVIEW

**Comment by eddie-c-davis** [comment]
launch jenkins

File: dawn/test/unit-test/dawn/Optimizer/Passes/samples/ReorderTest02.cpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include "gtclang_dsl_defs/gtclang_dsl.hpp"
+using namespace gtclang::dsl;
+
+stencil Test {
+  storage field_a0, field_a1, field_a2;
+  storage field_b0, field_b1, field_b2;
+
+  Do {
+    vertical_region(k_end, k_start) field_b1 = field_b0;
```
Comment (twicki) [inline dawn/test/unit-test/dawn/Optimizer/Passes/samples/ReorderTest02.cpp]: can we add `{}` here? I know that this notation technically works but the other one is cleaner in my eyes

Comment (eddie-c-davis) [reply dawn/test/unit-test/dawn/Optimizer/Passes/samples/ReorderTest02.cpp]: Done!

**Comment by eddie-c-davis** [comment]
launch jenkins

**Pull request merged**